src/Service/RequestService.php

```diff
@@ -784,6 +798,130 @@ public function proxyHandler(array $data, array $configuration, ?Source $proxy =
+            if (($usedSourceIds !== [] && in_array($proxy->getId()->toString(), $usedSourceIds) === false)
+                || ($excludedSourceIds !== [] && in_array($proxy->getId()->toString(), $excludedSourceIds) === true)
+            ) {
+                $proxies->remove($key);
+            }
+        }
+
+        unset($config['query']['_federalization_exclude_sources'], $config['query']['_federalization_use_sources']);
+
+        return $proxies;
+
+    }//end getFederationSources()
+
+    public function getFederationConfig(array $config): array
+    {
+        $config['timeout']     = 3;
+        $config['http_errors'] = true;
+
+        if (isset($config['query']['_federalization_timeout']) === true) {
+            $config['timeout'] = ($config['query']['_federalization_timeout'] / 1000);
+            unset($config['query']['_federalization_timeout']);
+        }
+
+        if (isset($config['query']['_federalization_ignore_error']) === true) {
+            $config['http_errors'] = $config['query']['_federalization_ignore_error'] === "false" ? true : false;
+            unset($config['query']['_federalization_ignore_error']);
+        }
+
+        return $config;
+
+    }//end getFederationConfig()
+
+    public function federationProxyHandler(Collection $proxies, string $path, array $config): Response
+    {
+        $this->requestTimes = [];
+
+        $proxies = $this->getFederationSources($config, $proxies);
+        $config  = $this->getFederationConfig($config);
+
+        $promises = [];
+        foreach ($proxies as $id => $proxy) {
+            $config['on_stats'] = function (TransferStats $stats) use ($id) {
+                $this->requestTimes[$id] = $stats->getTransferTime();
+            };
+
+            $promises[$id] = $this->callService->call($proxy, $path, 'GET', $config, true);
+        }
+
+        $responses = Utils::settle($promises)->wait();
+
+        $results['_sources'] = [];
+        $results['results']  = new ArrayCollection();
+        foreach ($responses as $id => $response) {
+            if ($response['state'] === 'rejected' && ($response['reason'] instanceof ConnectException || $config['http_errors'] === false)) {
+                continue;
+            } else if ($response['state'] === 'rejected' && ($response['reason'] instanceof ServerException || $response['reason'] instanceof ClientException)) {
+                throw $response['reason'];
+            }
+
+            $decoded            = $this->callService->decodeResponse($proxies[$id], $response['value']);
+            $decoded['results'] = array_map(
+                function (array $value) use ($proxies, $id) {
+                    $value['_source'] = $proxies[$id]->getId()->toString();
+                    return $value;
+                },
+                $decoded['results']
+            );
+            if ($response['value'] instanceof \GuzzleHttp\Psr7\Response === false) {
+                continue;
```

Comment:
Do we want to log something here 🤔

Reply:
Not necessarily, because ['value'] _always_ is a Response, this is merely for the comfort of the developer that the IDEs recognise ['value'] as a Response.